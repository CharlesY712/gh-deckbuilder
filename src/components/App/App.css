.deck-builder {
  display: flex;
  position: absolute;
  top: 50px;
  bottom: 50px;
}

#class-info {
  width: 220px;
  text-align: center;
  color: white;
  margin: auto;
}

#class-info img {
  width: 100%;
}

#class-info button {
  display: block;
  margin: 5px auto;
}

#class-info h2 {
  margin: 10px 10px 10px 15px;
}

#class-info h3 {
  display: inline-block;
  margin: 10px;
}

#class-info h4 {
  margin-bottom: 0;
}

#class-info button.inline-button {
  display: inline-block;
}