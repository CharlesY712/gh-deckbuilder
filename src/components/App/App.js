import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import {Header} from '../Header/Header';
import Footer from '../../containers/Footer/Footer';
import {Welcome} from '../Welcome/Welcome';
import DeckBuilder from '../../containers/DeckBuilder/DeckBuilder';
import './App.css';

class App extends Component {

  componentDidMount() {
    const background = require('../../images/background/background.png');
    document.body.style = `background-image: url(${background});`;
  }

  componentDidUpdate() {
    const background = require('../../images/background/background.png');
    document.body.style = `background-image: url(${background});`;
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Switch>
          <Route
            exact path="/"
            component={Welcome}
          />
          <Route
            path="/Brute"
            component={DeckBuilder}
          />
          <Route
            path="/Mindthief"
            component={DeckBuilder}
          />
          <Route
            path="/Cragheart"
            component={DeckBuilder}
          />
          <Route
            path="/Scoundrel"
            component={DeckBuilder}
          />
          <Route
            path="/Spellweaver"
            component={DeckBuilder}
          />
          <Route
            path="/Tinkerer"
            component={DeckBuilder}
          />
        </Switch>
        <Footer />
      </div>
    );
  }
}

export default App;
