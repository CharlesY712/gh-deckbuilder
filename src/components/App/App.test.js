import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App/>);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("componentDidUpdate", async () => {
    wrapper = shallow(<App/>, {disableLifecycleMethods: true});
    const update = wrapper.instance().componentDidUpdate();//eslint-disable-line
    const background = require('../../images/background/background.png');//eslint-disable-line
    const expectedStyle = {"background-image": "url(background.png)"};

    expect(document.body.style._values).toEqual(expectedStyle);
  });
});
