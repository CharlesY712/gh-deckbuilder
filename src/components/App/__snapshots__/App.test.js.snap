// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/Brute"
          />
          <Route
                    component={[Function]}
                    path="/Mindthief"
          />
          <Route
                    component={[Function]}
                    path="/Cragheart"
          />
          <Route
                    component={[Function]}
                    path="/Scoundrel"
          />
          <Route
                    component={[Function]}
                    path="/Spellweaver"
          />
          <Route
                    component={[Function]}
                    path="/Tinkerer"
          />
</Switch>,
        <withRouter(Connect(Footer)) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/Brute"
/>,
            <Route
              component={[Function]}
              path="/Mindthief"
/>,
            <Route
              component={[Function]}
              path="/Cragheart"
/>,
            <Route
              component={[Function]}
              path="/Scoundrel"
/>,
            <Route
              component={[Function]}
              path="/Spellweaver"
/>,
            <Route
              component={[Function]}
              path="/Tinkerer"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Brute",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Mindthief",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Cragheart",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Scoundrel",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Spellweaver",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/Tinkerer",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/Brute"
            />
            <Route
                        component={[Function]}
                        path="/Mindthief"
            />
            <Route
                        component={[Function]}
                        path="/Cragheart"
            />
            <Route
                        component={[Function]}
                        path="/Scoundrel"
            />
            <Route
                        component={[Function]}
                        path="/Spellweaver"
            />
            <Route
                        component={[Function]}
                        path="/Tinkerer"
            />
</Switch>,
          <withRouter(Connect(Footer)) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/Brute"
/>,
              <Route
                component={[Function]}
                path="/Mindthief"
/>,
              <Route
                component={[Function]}
                path="/Cragheart"
/>,
              <Route
                component={[Function]}
                path="/Scoundrel"
/>,
              <Route
                component={[Function]}
                path="/Spellweaver"
/>,
              <Route
                component={[Function]}
                path="/Tinkerer"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Brute",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Mindthief",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Cragheart",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Scoundrel",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Spellweaver",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/Tinkerer",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
